Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -89,7 +89,13 @@ class TopSitesPanel: UIViewController {
     let profile: Profile
 
     override func willRotateToInterfaceOrientation(toInterfaceOrientation: UIInterfaceOrientation, duration: NSTimeInterval) {
+        // ensure we always have the right data when we work out what our layout should be
+        self.refreshHistory(layout.thumbnailCount)
         layout.setupForOrientation(toInterfaceOrientation)
+    }
+
+    override func didRotateFromInterfaceOrientation(fromInterfaceOrientation: UIInterfaceOrientation) {
```

Comment:
willRotate/didRotate are now deprecated in iOS 8. Can we change these to use viewWillTransitionToSize instead? (https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIContentContainer_Ref/index.html#//apple_ref/occ/intfm/UIContentContainer/viewWillTransitionToSize:withTransitionCoordinator:)